File: study/lua/src/app/service/RecruitService.lua
local BaseService = require("app.service.BaseService")
local RecruitService = class("RecruitService",BaseService)
local FunctionConst = require("app.const.FunctionConst")

function RecruitService:ctor()
    RecruitService.super.ctor(self)
    self:start()
    self._hasNotice = false
end

function RecruitService:tick()
    --场景检测
    local runningSceneName = display.getRunningScene():getName() 
    if  runningSceneName ~= "main" then
        return
    end

    --判断是否开启功能
    local LogicCheckHelper = require("app.utils.LogicCheckHelper")
    local isOpen = LogicCheckHelper.funcIsOpened(FunctionConst.FUNC_WEEK_ACTIVITY)
    if not isOpen then
        return
    end

    local hasCount = G_UserData:getRecruitData():hasFreeCount()
    if not hasCount then
        return
    end

    if G_UserData:getRecruitData():hasFreeNormalCount() then
        if not self._hasNotice then
            G_SignalManager:dispatch(SignalConst.EVENT_RED_POINT_UPDATE)
            self._hasNotice = true
        end
    else
        self._hasNotice = false
    end
end



return RecruitService

File: study/ts/script/service/RecruitService.ts
import { BaseService } from "./BaseService";
import { G_SceneManager, G_UserData, G_SignalManager } from "../init";
import { LogicCheckHelper } from "../utils/LogicCheckHelper";
import { FunctionConst } from "../const/FunctionConst";
import { SignalConst } from "../const/SignalConst";

export class RecruitService extends BaseService {

    _hasNotice: boolean;
    constructor() {
        super();
        this.start();
        this._hasNotice = false;
    }
    public tick() {
        let runningSceneName = G_SceneManager.getRunningScene().getName();
        if (runningSceneName != 'main') {
            return;
        }
        let [ isOpen ] = LogicCheckHelper.funcIsOpened(FunctionConst.FUNC_WEEK_ACTIVITY);
        if (!isOpen) {
            return;
        }
        let hasCount = G_UserData.getRecruitData().hasFreeCount();
        if (!hasCount) {
            return;
        }
        if (G_UserData.getRecruitData().hasFreeNormalCount()) {
            if (!this._hasNotice) {
                G_SignalManager.dispatch(SignalConst.EVENT_RED_POINT_UPDATE);
                this._hasNotice = true;
            }
        } else {
            this._hasNotice = false;
        }
    }
}
